### PR TITLE
Fix for randomly failing XHMM-related tests

### PR DESCRIPTION
**Pull request by mbabadi**
Still not sure why the tests failed randomly! all XHMM-related tests use their own RNG with fixed seeds and there are no RNG calls in any parallel streams. Therefore, the randomly generated test data must be identical and fully deterministic across all runs. However, it did not appear to be the case! some test runs triggered a bug in HMMPostProcessor (see below) and some runs didn't. I removed a few unnecessary RNGs and the issue is not reproducible anymore. In particular, both XHMMModel and XHMMEmissionProbabilityCalculator had their own RNG but then again, if the tests are run in a deterministic order, it shouldn't matter. The good news is the bug in HMMPostProcessor is fixed; the bad news is, I still don't know why the tests were not deterministic. I bet the failing issue is (magically!) fixed as a result of pulling out the RNG from XHMMModel and XHMMEmissionProbabilityCalculator. If it occurs again, I'll investigate more.

- fixed a bug in HMMPostProcessor that required all samples to be queried in the given list of genotyping segments every time (origin of the failing tests: sometimes the randomly generated genotyping segments contained fewer samples than all samples available for genotyping)
- got rid of the unnecessary RNG in XHMMModel to make it stateless (sampling requires an external RNG)
- also made XHMMEmissionProbabilityCalculator stateless (sampling requires an external RNG)
- truncated the target list used in XHMM integration tests (cuts down the test time by a factor of 10)

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/exome/germlinehmm/xhmm/XHMMEmissionProbabilityCalculator.java [hunk 1]
```diff
@@ -10,30 +10,28 @@
...
         this.deletionMean = ParamUtils.isNegativeOrZero(deletionMean, "Deletion coverage shift must be negative.");
         this.duplicationMean = ParamUtils.isPositiveOrZero(duplicationMean, "Duplication coverage shift must be positive");
         emissionStandardDeviation = ParamUtils.isPositive(emissionStdDev, "Emission standard deviation must be positive");
-        this.rng = rng;
     }
 
     @Override
     public double logLikelihood(@Nonnull final XHMMEmissionData emissionData, final double copyRatio, final Target target) {
-        return new NormalDistribution(rng, getEmissionMean(copyRatio), emissionStandardDeviation)
+        return new NormalDistribution(getEmissionMean(copyRatio), emissionStandardDeviation)
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/germlinehmm/xhmm/XHMMEmissionProbabilityCalculator.java]: According to the documentation, passing null as the rng to NormalDistribution will avoid some overhead.

**Comment by samuelklee** [comment]
Looks good to me except for one minor thing.  Hopefully this does the trick!  Thanks also for cutting down on the test runtime.

**Comment by droazen** [comment]
@mbabadi Can you rebase this onto the latest master (which has a fix for the travis failures) and merge ASAP? Thanks!

File: src/main/java/org/broadinstitute/hellbender/tools/exome/germlinehmm/xhmm/XHMMEmissionProbabilityCalculator.java (hunk 1 above)
Comment (mbabadi) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/germlinehmm/xhmm/XHMMEmissionProbabilityCalculator.java]: done.

**Comment by droazen** [comment]
M2 WDL tests had a transient failure on this branch -- re-running them.

**Comment by droazen** [comment]
Merging this despite the unrelated "No space left on device" error in the M2 WDL tests, which is definitely unrelated to this change.

**Pull request merged**